pytorch_lightning/trainer/training_loop.py

```diff
@@ -513,12 +513,11 @@ def run_training_epoch(self):
             # -----------------------------------------
             should_check_val = self.should_check_val_fx(batch_idx, is_last_batch)
             if should_check_val:
+                self.trainer.validating = True
```

Comment:
should this be inside `run_evaluation` itself?

Reply:
I don't think so because `run_evaluation` does not know which stage/state is calling it.

It could be from `.test()`, `.validate()` or the validation part of `.fit()`

Reply:
So it's tricky to define the entry state and exit state from inside

Reply:
you have added trainer state now right (fitting, validating)? can those be used?

Reply:
No, `TrainerState` is used just for high-level user logic (did the user call `trainer.fit` or `trainer.validate`?) whereas `RunningStage` is the particular internal stage the trainer is at (training, validating...)

So `run_evaluation` works with the second one, does not know about the first one

Notice that most of the times `TrainerState` and `RunningStage` match (for example, `trainer.tune` only runs the `tuning` stage). But it is necessary to differentiate if we are validating because the user called `trainer.fit` or if it is because the user called `trainer.validate`